pkgs/development/libraries/atk/default.nix

```diff
@@ -39,15 +39,13 @@ stdenv.mkDerivation rec {
     fixDarwinDylibNames
   ];
 
+  buildInputs = [ gobject-introspection ];
```

Comment:
Why add this to `buildInputs`? It should not be a runtime dependency.

Reply:
libgirepository can be used at runtime, right?

Reply:
Yes, but ATK should not depend on `libgirepository`.

Reply:
Then it shouldn't be in buildInputs :)

Reply:
For some reason it then does require it at runtime, or the meson file is wrong.

Reply:
Perhaps it needs `native=True` here: https://github.com/mesonbuild/meson/blob/16ed82a62f535475b0eff6acc2d7c34c5265c16f/mesonbuild/modules/gnome.py#L772

Reply:
i applied

```diff
diff --git a/pkgs/development/tools/build-managers/meson/default.nix b/pkgs/development/tools/build-managers/meson/default.nix
index f62bb22314d..b46f918b9c3 100644
--- a/pkgs/development/tools/build-managers/meson/default.nix
+++ b/pkgs/development/tools/build-managers/meson/default.nix
@@ -71,6 +71,11 @@ python3.pkgs.buildPythonApplication rec {
     ./fix-gtkdoc-when-using-multiple-apple-frameworks.patch
   ];
 
+  postPatch = ''
+    substituteInPlace mesonbuild/modules/gnome.py \
+      --replace "(state, 'gobject-introspection-1.0')" "(state, 'gobject-introspection-1.0', native = True)"
+  '';
+
   setupHook = ./setup-hook.sh;
 
   # Meson included tests since 0.45, however they fail in Nixpkgs because they
```

to meson because of some projects using the wrong gi when build == host != target and gobject-introspection was in buildInputs

that fixed them

gobject-introspection still has to be in buildInputs

otherwise

```
json-glib-aarch64-unknown-linux-gnu> /nix/store/s9m5ww62bsyx0jzylndxicvkmdn7x5bj-aarch64-unknown-linux-gnu-binutils-2.38/bin/aarch64-unknown-linux-gnu-ld: skipping incompatible /nix/store/c79djpxhga419drlbf6v850pvpjfqsgl-gobject-introspection-wrapped-1.72.0/lib/libgirepository-1.0.so when searching for -lgirepository-1.0
json-glib-aarch64-unknown-linux-gnu> /nix/store/s9m5ww62bsyx0jzylndxicvkmdn7x5bj-aarch64-unknown-linux-gnu-binutils-2.38/bin/aarch64-unknown-linux-gnu-ld: cannot find -lgirepository-1.0: No such file or directory
json-glib-aarch64-unknown-linux-gnu> /nix/store/s9m5ww62bsyx0jzylndxicvkmdn7x5bj-aarch64-unknown-linux-gnu-binutils-2.38/bin/aarch64-unknown-linux-gnu-ld: skipping incompatible /nix/store/c79djpxhga419drlbf6v850pvpjfqsgl-gobject-introspection-wrapped-1.72.0/lib/libgirepository-1.0.so when searching for -lgirepository-1.0
```

when gi is in buildInputs

```
⋮ 245│/nix/store/s9m5ww62bsyx0jzylndxicvkmdn7x5bj-aarch64-unknown-linux-gnu-binutils-2.38/bin/aarch64-unknown-linux-gnu-ld: skipping incompatibl
e /nix/store/c79djpxhga419drlbf6v850pvpjfqsgl-gobject-introspection-wrapped-1.72.0/lib/libgirepository-1.0.so when searching for -lgirepository-1.0
```

Reply:
But [`libgirepostiory`](https://gi.readthedocs.io/en/latest/writingbindings/libgirepository.html), as I understand it, is for runtime loading of typelibs. I still do not see any reason for libraries just providing typelibs to need it – it should only be necessary for stuff like `pygobject`.